src/shared/import-util.c

```diff
@@ -134,19 +134,71 @@ static const char* const import_verify_table[_IMPORT_VERIFY_MAX] = {
 
 DEFINE_STRING_TABLE_LOOKUP(import_verify, ImportVerify);
 
+static const char* const import_compress_type_table[_IMPORT_COMPRESS_TYPE_MAX] = {
+        [IMPORT_COMPRESS_UNKNOWN] = "unknown",
+        [IMPORT_COMPRESS_UNCOMPRESSED] = "uncompressed",
+        [IMPORT_COMPRESS_XZ] = "xz",
+        [IMPORT_COMPRESS_GZIP] = "gzip",
+#if HAVE_BZIP2
+        [IMPORT_COMPRESS_BZIP2] = "bzip2",
+#endif
```

Comment:
i think the ifdeffery should be dropped here. we should recognize the suffix even if bzip2 support is off, and then fail when actually an attempt is made to use this

Reply:
Yes, fair enough. I have also redid the machinectl logic so that it passes through the compression level unmodified, and the decision is made in `systemd-export`. This way the separation of concepts is upheld.